src/logging.rs

```diff
@@ -529,6 +760,65 @@ fn update_log_ui(
                 ))
                 .id();
 
+            // Insert the relevant log line marker and set visibility based on the log level.
+            match *e.metadata.level() {
+                tracing::Level::ERROR => commands.entity(child_entity).insert((
+                    ErrLogLineMarker,
+                    Style {
+                        display: if log_viewer_res.error_visible {
+                            Display::Flex
+                        } else {
+                            Display::None
+                        },
+                        ..default()
+                    },
+                )),
+                tracing::Level::WARN => commands.entity(child_entity).insert((
+                    WarnLogLineMarker,
+                    Style {
+                        display: if log_viewer_res.warn_visible {
+                            Display::Flex
+                        } else {
+                            Display::None
+                        },
+                        ..default()
+                    },
+                )),
+                tracing::Level::INFO => commands.entity(child_entity).insert((
+                    InfoLogLineMarker,
+                    Style {
+                        display: if log_viewer_res.info_visible {
+                            Display::Flex
+                        } else {
+                            Display::None
+                        },
+                        ..default()
+                    },
+                )),
+                tracing::Level::DEBUG => commands.entity(child_entity).insert((
```

Comment:
please add a helper method here to reduce duplication, it would take the commands, child_entity, marker, visibility params as input.